### DISCUSSION TITLE
Seems to only half work

### DISCUSSION THREAD

**Discussion by scubanarc** [torum/MPDCtrl#3] [General]
I have a working mpd server and I can control it with mpc, ncmpcpp, and other clients. MPDCtrl can sort of half control it. My server does not have a password.

What works:
- Current song shows at the top
- Prev, play/pause, next, and volume all work
- Search appears to show all files
- Playlists show, but...

What doesn't work:
- My playlists are showing as between 1 and 5 files, even though they are 100's of files long.
- My queue only shows 1 file, when I actually have about 3000 files queued.
- No album art is showing.

I have tried the "readpicture" setting.

I have enabled debug logs and they look normal.

I have enabled MPDCtrl_errors.txt and the file never appears.

Any ideas?

**Comment by torum** [comment]
Hi. Sorry for the delay and thank you for your participation.

At the moment, I have no idea what is going on. However, it sounds kind of similar to what happens when MPD server is very old. Could you tell me what version of MPD server you are using? I have tested MPDCtrl against latest version of MPD on Windows 10 and Ubuntu Linux with no problem at all.

PS. Also, please make sure you are using the latest version of MPDCtrl as well.

Thank you.

**Comment by scubanarc** [comment]
I am running Ubuntu:

> VERSION="20.04.1 LTS (Focal Fossa)"

The default version of mpd is:

> Music Player Daemon 0.21.20 (0.21.20)

I followed the instructions here to compile from source:

https://www.musicpd.org/doc/html/user.html

I am now running:

> Music Player Daemon 0.22.6 (0.22.6)

Which is the latest.

It's definitely working better, but there are some playlist and current queue bugs. The album art is now showing. However, my playlists are still showing as very short. They show fine in ncmpcpp and in myMPD (docker container with http front end). My current queue is 10k items long but I only see 13 in MPDCtrl.

If I exit and restart, I can see the following happen in the bottom RIGHT window:

>MPDCtrl: status
>MPD: (returns status)
>MPDCtrl: currentsong
>MPD: (returns current song info)
>MPDCtrl: playlistinfo
>MPD: (returns giant playlist)
>MPDCtrl: listplaylists
>MPD: (returns all playlists)

The current queue is not parsed correctly because it's only showing 13 entries (this time, it's different sometimes)

More info... if I click a playlist, then in the bottom LEFT window:

>MPDCtrl: listplaylistinfo "my-playlist-name"
>MPD: (returns giant playlist)

But MPDCtrl only shows a few of the files. The last file that WORKS is:

><<<<file: The Black Keys/El Camino/The Black Keys - Run Right Back.mp3
><<<<Last-Modified: 2019-09-15T23:25:02Z
><<<<Artist: The Black Keys
><<<<AlbumArtist: The Black Keys
><<<<Title: Run Right Back
><<<<Album: El Camino
><<<<Track: 6
><<<<Date: 2011-12-02
><<<<Date: 2011
><<<<Genre: Rock
><<<<Disc: 1
><<<<Time: 197
><<<<duration: 197.093

The first file that DOES NOT WORK / BREAKS MPDCtrl is:

><<<<file: Deadmau5/- album title goes here -/deadmau5 - October.mp3
><<<<Last-Modified: 2019-09-15T23:13:36Z
><<<<Artist: deadmau5
><<<<AlbumArtist: deadmau5
><<<<Title: October
><<<<Album: > album title goes here <
><<<<Track: 10
><<<<Genre: Electronic
><<<<Time: 444
><<<<duration: 443.951

You think that the "greater than" and "less than" in the Album: field are breaking the parse?

**Reply by torum** [reply]
Thank you for your info. I appreciate very much.
I think you are right. It seems like a parsing problem. Although "greater than" and "less than" characters shouldn't cause any problems, ":"(colons) or carriage returns might cause problems since those are delimiters. 

I need to test with those files. Please give me some time. Thank you!

Edit: "> album title goes here <" did not break anything for me. A colon didn't do anything either...

**Comment by torum** [comment]
Hi, I'm sorry for the delay. 
I receive a new bug report very similar to yours in issue page a few days ago.
#4 

I think I fixed the bug. Hope you try the new version (v3.0.7). Thank you!

**Reply by scubanarc** [reply]
I updated to v3.0.7 today and yes, it fixed all of my problems. It appears to work perfectly for me now.

Thanks, I really like the interface and I hope you keep developing it.
